### PR TITLE
add validate & validateFirst to ZSTM companion

### PR DESCRIPTION
**Pull request by sken77**
https://github.com/zio/zio/issues/2802

### CODE REVIEW

**Comment by sken77** [comment]
@adamgfraser any ideas on this error?

File: core/shared/src/main/scala/zio/stm/STM.scala [hunk 1]
```diff
@@ -277,6 +277,22 @@ object STM {
   val unit: STM[Nothing, Unit] =
     ZSTM.unit
 
+  /**
+   * @see See [[zio.stm.ZSTM.validate]]
+   */
+  def validate[R, E, A, B](
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/stm/STM.scala]: There is no `R` in `STM`.

File: core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala [hunk 2]
```diff
@@ -1045,6 +1045,50 @@ object ZSTMSpec extends ZIOBaseSpec {
         assertM(tx.commit)(equalTo(1))
       }
     ),
+    suite("ZSTM validate")(
+      testM("returns all errors if never valid") {
+        val in  = List.fill(10)(0)
+        val res = ZSTM.validate(in)(a => ZSTM.failNow(a))
```
Comment (mijicd) [inline core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala]: Replace with calls to STM, such that no environment is needed.

File: core/shared/src/main/scala/zio/stm/STM.scala (hunk 1 above)
Comment (sken77) [reply core/shared/src/main/scala/zio/stm/STM.scala]: does, partition above also needs then?

Comment (sken77) [reply core/shared/src/main/scala/zio/stm/STM.scala]: oh i see the commit now

File: core/shared/src/main/scala/zio/stm/STM.scala [hunk 3]
```diff
@@ -279,6 +279,22 @@ object STM {
...
+  )(f: A => STM[E, B])(implicit ev: CanFail[E]): STM[::[E], List[B]] =
+    ZSTM.validate(in)(f)
+
+  /**
+   * @see See [[zio.stm.ZSTM.validateFirst]]
+   */
+  def validateFirst[E, A, B](
+    in: Iterable[A]
+  )(f: A => STM[E, B])(implicit ev: CanFail[E]): STM[List[E], B] =
+    ZSTM.foreach(in)(f(_).flip).flip
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/stm/STM.scala]: Don't we want this to just be an alias for the method on `ZSTM`?

Comment (sken77) [reply core/shared/src/main/scala/zio/stm/STM.scala]: we do. you can slap me for that one

**Pull request merged**